### ISSUE TITLE
Authentication for Client.Legacy is only URL-Based

### ISSUE THREAD

**Issue by Wawszczak** [influxdata/influxdb-client-csharp#84]
Unfortunatelly, URL-based authentication mechanizm is not recomended by InfluxDB documentation.
Moreover, it is unsecure because URLs are visible in every log of network device.
Is there any possibility to add HTTP Basic Auth scheme to the list of possible options?

**Comment by bednar** [comment]
@Wawszczak thanks! we will take a look!

**Comment by bednar** [comment]
@rolincova, reuse an implementation from v2 https://github.com/influxdata/influxdb-client-csharp/blob/dea0c9d56ad62bf4d8a7a58d68b4b3f914d87777/Client/Internal/ApiClient.cs#L54

**Comment by bednar** [comment]
Hi @Wawszczak,

The issue is fixed in [1.8.0 milestone](https://github.com/influxdata/influxdb-client-csharp/milestone/11). 

```c#
// client creation
var options = new FluxConnectionOptions("http://127.0.0.1:8086", "my-user", "my-password".ToCharArray(),
    FluxConnectionOptions.AuthenticationType.BasicAuthentication);

var fluxClient = FluxClientFactory.Create(options);

fluxClient.QueryAsync(...)
...
```

details - https://github.com/influxdata/influxdb-client-csharp/tree/master/Client.Legacy#basic-authentication

If you would like to use a preview version use our nightly build: [InfluxDB.Client 1.8.0-dev.669](https://www.nuget.org/packages/InfluxDB.Client/1.8.0-dev.669).

Regards
